src/string_object.cpp

```diff
@@ -579,6 +579,29 @@ size_t StringObject::char_count(Env *env) const {
     return char_count;
 }
 
+Value StringObject::setbyte(Env *env, Value index_obj, Value value_obj) {
+    assert_not_frozen(env);
+
+    nat_int_t index = IntegerObject::convert_to_nat_int_t(env, index_obj);
+    nat_int_t value = IntegerObject::convert_to_nat_int_t(env, value_obj);
+    nat_int_t length = static_cast<nat_int_t>(m_string.length());
+
+    if (index + length < 0) {
+        env->raise("IndexError", "index {} out of string", index);
+    }
```

Comment:
I don't understand this check... Is this supposed to check for a negative `index`? If so, why add `length` to it?